fendermint/actors/blobs/shared/src/state.rs

```diff
@@ -238,6 +298,41 @@ pub struct Blob {
     pub status: BlobStatus,
 }
 
+/// The return type used for Blob.
+#[derive(Debug, Serialize_tuple, Deserialize_tuple)]
+pub struct BlobInfo {
+    /// The size of the content.
+    pub size: u64,
+    /// Blob metadata that contains information for blob recovery.
+    pub metadata_hash: Hash,
+    /// Active subscribers (accounts) that are paying for the blob to expiry.
+    pub subscribers: HashMap<SubscriptionId, ChainEpoch>,
+    /// Blob status.
+    pub status: BlobStatus,
+}
+
+impl BlobInfo {
+    /// Returns [`BlobInfo`] from [`Blob`].
+    /// TODO: HAMTs should carry max expiry such that we don't full scan here.
```

Comment:
we do paginate the expiry now so this TODO could be removed, right? or is this a TODO for something else?

Reply:
this is new, for the building of `BlobInfo`